### PR TITLE
feat: add an annotation for ignoring URI case in HTTP match

### LINKED ISSUE #163: Implement annotation to support case-insensitive path matching
## How could it be?
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    higress.io/ignore-path-case: "true"
  name: foo
spec:
  ingressClassName: higress  
  rules:
  - http:
      paths:
      - pathType: Exact
        path: "/abc"
...
```

## Other related information
[1. Annotation example](https://github.com/alibaba/higress/blob/main/pkg/ingress/kube/annotations/cors.go)
[2. Istio api conversion](https://github.com/alibaba/higress/blob/main/pkg/ingress/config/ingress_config.go#L331)
[3. Istio VitrualService's MatchRequest api](https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPMatchRequest)

### PR DESCRIPTION
**Pull request by Charlie17Li**
Signed-off-by: charlie <qianglin98@qq.com>

### Ⅰ. Describe what this PR did

add an annotation for ignoring URI case in HTTP match

### Ⅱ. Does this pull request fix one issue?

fixes #163

### CODE REVIEW

**Comment by johnlanni** [comment]
@Charlie17Li  Could you add an e2e test case for it?

File: test/ingress/conformance/tests/httproute-ignore-case-match.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	"github.com/alibaba/higress/test/ingress/conformance/utils/http"
+	"github.com/alibaba/higress/test/ingress/conformance/utils/suite"
+)
+
+func init() {
+	HigressConformanceTests = append(HigressConformanceTests, HTTPRouteIgnoreCaseMatch)
+}
+
+var HTTPRouteIgnoreCaseMatch = suite.ConformanceTest{
```
Comment (Xunzhuo) [inline test/ingress/conformance/tests/httproute-ignore-case-match.go:28]: please add HTTPRouteIgnoreCaseMatch to [e2e_test.go](https://github.com/alibaba/higress/blob/main/test/ingress/e2e_test.go#L53)

**Review by Xunzhuo — CHANGES_REQUESTED** [review]

File: test/ingress/conformance/tests/httproute-ignore-case-match.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	ShortName:   "HTTPRouteIgnoreCaseMatch",
+	Description: "A Ingress in the higress-conformance-infra namespace that ignore URI case in HTTP match",
```
Comment (Xunzhuo) [inline test/ingress/conformance/tests/httproute-ignore-case-match.go:30]: nit: ignore -> ignores

File: test/ingress/conformance/tests/httproute-ignore-case-match.go (hunk 1 above)
Comment (Xunzhuo) [reply test/ingress/conformance/tests/httproute-ignore-case-match.go:28]: Because this could be referred as a separated pkg to use.

**Review by Xunzhuo — APPROVED** [review]
LGTM, thanks for your contribution @Charlie17Li !

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**